addons/hr_holidays/models/hr_leave.py

```diff
@@ -1602,15 +1601,16 @@ def activity_update(self):
                 to_clean |= holiday
             elif holiday.state == 'confirm':
                 user_ids = holiday.sudo()._get_responsible_for_approval().ids or self.env.user.ids
-                for user_id in user_ids:
-                    activity_vals.append({
-                        'activity_type_id': self.env.ref('hr_holidays.mail_act_leave_approval').id,
-                        'automated': True,
-                        'note': note,
-                        'user_id': user_id,
-                        'res_id': holiday.id,
-                        'res_model_id': self.env.ref('hr_holidays.model_hr_leave').id,
-                    })
+                if holiday.holiday_status_id.responsible_ids:
+                    for user_id in user_ids:
+                        activity_vals.append({
+                            'activity_type_id': self.env.ref('hr_holidays.mail_act_leave_approval').id,
```

Comment:
Even though it's the existing code, the 2 `ref()` can be moved outside of the for loop to speed things up a little bit